pyccel/codegen/printing/ccode.py

```diff
@@ -2184,6 +2187,14 @@ def _print_Assign(self, expr):
         rhs = self._print(expr.rhs)
         return prefix_code+'{} = {};\n'.format(lhs, rhs)
 
+    def _print_SetPop(self, expr):
+        var_type = self.get_declare_type(expr.set_variable)
+        element_type = self.get_c_type(expr.set_variable.class_type.element_type)
+        self.add_import(Import(f'Set_pop_macro/{var_type}/{element_type}', Module(f'Set_pop_macro/{var_type}/{element_type}', (), ())))
+        self.add_import(Import('STC_Extensions/Set_extensions', Module('STC_Extensions/Set_extensions', (), ())))
```

Comment:
Does this require the imports to be printed in order?

Reply:
If you're talking about order of defining the macros and include of `Set_extensions`, yes the file need the `i_type` and `i_key`.

Reply:
I mean what happens if the imports are stored in a set and they end up out of order.

Reply:
I think I understand what you meant, so I tried this solution : 82aa8fd
 If you have a better idea, that would be good.

Reply:
This looks safer :+1: don't you need a scope guard though?
```c
#ifdef ...
```